kairon/shared/data/collection_processor.py

```
@@ -310,5 +310,60 @@
+    @staticmethod
+    def validate_collection_name(collection_name):
+        try:
+            if not collection_name or not collection_name.strip():
+                raise ValueError("Collection name cannot be empty.")
+            if len(collection_name) > 64:
+                raise ValueError("Collection name cannot exceed 64 characters.")
+            if not re.match(r"^[A-Za-z][A-Za-z0-9_-]*$", collection_name):
+                raise ValueError(
+                    "Collection name must start with a letter and contain only letters, numbers, underscores, or hyphens.")
+        except Exception as e:
+            raise HTTPException(status_code=422, detail=str(e))
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)